src/QueryBuilder.js

```diff
@@ -1566,6 +1568,191 @@ QueryBuilder.prototype.returning = queryMethod('returning');
+ */
+QueryBuilder.prototype.whereJsonIsObject = function (fieldExpression) {
+  return this.whereJsonSupersetOf(fieldExpression, {});
+};
+
+/**
+ * Match if one of given strings is found from json object key(s) or array items.
+ *
+ * @param {FieldExpression} fieldExpression
+ * @param {String|Array.<String>} keys Strings that are looked from object or array.
+ * @returns {QueryBuilder}
+ */
+/* istanbul ignore next */ // TODO: remove this when tests are enabled and knex bug #519 is fixed
+QueryBuilder.prototype.whereJsonHasAny = function (fieldExpression, keys) {
+  throw new Error("Disabled because of knex issue #519.");
+  // return whereJsonFieldRightStringArrayOnLeft(this, fieldExpression, '?|', keys);
+};
+
+/**
+ * Match if all strings are found from json object key(s) or array items.
+ *
+ * @param {FieldExpression} fieldExpression
+ * @param {String|Array.<String>} keys Strings that are looked from object or array.
+ * @returns {QueryBuilder}
+ */
+/* istanbul ignore next */ // TODO: remove this when tests are enabled and knex bug #519 is fixed
+QueryBuilder.prototype.whereJsonHasAll = function (fieldExpression, keys) {
+  throw new Error("Disabled because of knex issue #519.");
+  // return whereJsonFieldRightStringArrayOnLeft(this, fieldExpression, '?&', keys);
+};
+
+/**
+ * Match if json field value comparison match.
+ *
+ * Value may be number, string, null, boolean and referred json field is converted
+ * to TEXT, NUMERIC or BOOLEAN sql type for comparison.
+ *
+ * If left hand field does not exist rows appear IS null so if one needs to get only
+ * rows, which has key and it's value is null one may use e.g.
+ * `.whereJsonSupersetOf("column", { field: null })` or check is key exist and
+ * then `.whereJsonField('column.field', 'IS', null)`
+ *
+ * For testing against objects or arrays one should see tested with whereJsonEqual,
+ * whereJsonSupersetOf and whereJsonSubsetOf methods.
+ *
+ * @param {FieldExpression} fieldExpression Expression pointing to certain value.
+ * @param {String} operator SQL comparator usually `<`, `>`, `<>`, `=` or `!=`
+ * @param {Boolean|Number|String|null} value Value to which field is compared to.
+ * @returns {QueryBuilder}
+ */
+QueryBuilder.prototype.whereJsonField = function (fieldExpression, operator, value) {
+  var knex = this._modelClass.knex();
+  var fieldReference = parseFieldExpression(fieldExpression, true);
+  // json type comparison takes json type in string format
+  var cast;
+  var escapedValue = knex.raw(" ?", [value]);
+  if (_.isNumber(value)) {
+    cast = "::NUMERIC";
+  } else if (_.isBoolean(value)) {
+    cast = "::BOOLEAN";
+  } else if (_.isString(value)) {
+    cast = "::TEXT";
+  } else if (_.isNull(value)) {
+    cast = "::TEXT";
+    escapedValue = 'NULL';
+  } else {
+    throw new Error("Value must be string, number, boolean or null.");
+  }
+  return this.whereRaw(["(", fieldReference, ")", cast, " ", operator," ", escapedValue].join(""));
```

Comment:
Maybe that `IN` support needs to be added later... it will need some designing first, since casting in that case is not that straight forward. I'll add separate feature issue about it.